histomicstk/features/ComputeGradientFeatures.py

```diff
@@ -0,0 +1,68 @@
+import numpy as np
+from scipy.stats import entropy, skew, kurtosis
+from skimage.feature import canny
+
+
+def ComputeGradientFeatures(I, Coords, Diff=[], C=[]):
+    """
+    Calculates gradient features from an intensity image.
+
+    Parameters
+    ----------
+    I : array_like
+        A M x N intensity image.
+    Coords : array_like
+        A T x 2 coordinate list of a region.
+    Diff : array_like
+        A M x N gradient difference. Default value = [].
+    C : array_like
+        A M x N canny filtered mask. Default value = [].
+
+    Returns
+    -------
+    MeanGradMag : float
+        Mean of gradient data.
+    StdGradMag : float
+        Standard deviation of gradient data.
+    EntropyGradMag : float
+        Entroy of gradient data.
+    EnergyGradMag : float
+        Energy of gradient data.
+    SkewnessGradMag : float
+        Skewness of gradient data. Value is 0 when all values are equal.
+    KurtosisGradMag : float
+        Kurtosis of gradient data. Value is -3 when all values are equal.
+    SumCanny : float
+        Sum of canny filtered gradient data.
+    MeanCanny : float
+        Mean of canny filtered gradient data.
+
+    References
+    ----------
+    .. [1] Daniel Zwillinger and Stephen Kokoska. "CRC standard probability
+    and statistics tables and formulae," Crc Press, 1999.
+    """
+
+    if not len(Diff):
+        Gx, Gy = np.gradient(I)
+        Diff = np.sqrt(Gx**2 + Gy**2)
```

Comment:
It probably took more time because it applies smoothing before computing the gradient. A little bit of smoothing `sigma = 1` or `sigma=2` would give us some resilience against any noise.